### ISSUE TITLE
Should class has-background be in blocks where you can add background color

### ISSUE THREAD

**Issue by samikeijonen** [WordPress/gutenberg#11172]
At the moment there is  class `has-background` in paragraph block. But I don't see it in other blocks where you can add background color. For example in Media Text block.

We could target something like `[class$="background-color"]`. But I'd find more helpful to be consistent on outputted class names. 

Suggestion: Every block that supports bg color should have `has-background` class.

**Comment by rleeson** [comment]
Tested adding background colors to Pull Quotes, Media & Text, and Paragraphs. Confirmed Paragraphs add both the **has-background** and **has-{colorname}-background** classes added to the block wrapper, while Pull Quotes and Media & Text adds only the **has-{colorname}-background** class.

Agree this seems a good enhancement to add for consistency in styling.

**Confirmed with**
- WordPress 4.9.8
- Gutenberg 4.1.1

**Comment by MichaelRise** [comment]
Simple solution at this moment - [Selectors Level 3](https://www.w3.org/TR/selectors-3/#selectors)
Maybe this will help someone.
```
.wp-block-test[class*="-background-color"] {
[...]
[...]
}
```

**Comment by samikeijonen** [comment]
Thanks @MichaelRise, I also had similar example in the issue.

**Comment by chrisvanpatten** [comment]
I might try to pick this up this week.

**Comment by chrisvanpatten** [comment]
Addressed in #11228!

**Comment by justintadlock** [comment]
Of note, you cannot simply target `[class*=background-color]`.  You must also target `[style*=background-color]` when the user selects a color the theme hasn't registered.

Here's a look at something I'm having to do because of this:

```css
.alignfull[style*=background-color] + .alignfull[style*=background-color],
.alignfull[style*=background-color] + .alignfull[class*=background-color],
.alignfull[class*=background-color] + .alignfull[class*=background-color],
.alignfull[class*=background-color] + .alignfull[style*=background-color] {
	margin-top: -2em;
	border-top: 0;
}
```

With the `.has-background` class, that could be reduced and save a bit of CSS.

**Comment by tellthemachines** [comment]
Closing as this was fixed in #17529.
